pkg/prometheus/operator.go

```diff
@@ -1702,19 +1702,21 @@ func createSSetInputHash(p monitoringv1.Prometheus, c operator.Config, ruleConfi
 	hash, err := hashstructure.Hash(struct {
 		PrometheusLabels      map[string]string
 		PrometheusAnnotations map[string]string
-		PrometheusSpec        monitoringv1.PrometheusSpec
-		Config                operator.Config
-		StatefulSetSpec       appsv1.StatefulSetSpec
-		RuleConfigMaps        []string `hash:"set"`
-		Assets                []string `hash:"set"`
+		//PrometheusSpec        monitoringv1.PrometheusSpec
```

Comment:
We may want to keep the PrometheusSpec field in the hash computation for now but I wanted to give the CI a try with only the generation field.